pkg/tsdb/cloudwatch/routes/metrics.go

```diff
@@ -0,0 +1,44 @@
+package routes
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models/request"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/services"
+)
+
+func MetricsHandler(pluginCtx backend.PluginContext, clientFactory models.ClientsFactoryFunc, parameters url.Values) ([]byte, *models.HttpError) {
+	metricsRequest, err := request.GetMetricsRequest(parameters)
+	if err != nil {
+		return nil, models.NewHttpError("error in MetricsHandler", http.StatusBadRequest, err)
+	}
+
+	service, err := newListMetricsService(pluginCtx, clientFactory, metricsRequest.Region)
+	if err != nil {
+		return nil, models.NewHttpError("error in MetricsHandler", http.StatusInternalServerError, err)
+	}
+
+	var metrics []*models.Metric
+	switch metricsRequest.Type() {
+	case request.AllMetricsRequestType:
+		metrics = services.GetAllHardCodedMetrics()
+	case request.MetricsByNamespaceRequestType:
+		metrics, err = services.GetHardCodedMetricsByNamespace(metricsRequest.Namespace)
+	case request.CustomNamespaceRequestType:
+		metrics, err = service.GetMetricsByNamespace(metricsRequest.Namespace)
+	}
```

Comment:
Not sure if Sarah has already mentioned this, but I think tests at this routes level for the different results of the switch case could be interesting.

(same for dimension_keys)